### PR TITLE
Implemented Protean and Libero

### PR DESCRIPTION
**Pull request by takeyourt1me**
Adds functionality for Libero and Protean abilities

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,4 +35,5 @@ src/data/battle-anim-raw-data*.ts
 src/data/battle-anim-data.ts
 src/overrides.ts
 
-coverage
\ No newline at end of file
+coverage
+src/overrides.ts
```
Comment (Flashfyre) [inline .gitignore:39]: What's the purpose of this duplicate line?

**Review by Flashfyre — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (takeyourt1me) [reply .gitignore:39]: Im not sure, must have accidentally left it in

File: src/field/pokemon.ts [hunk 2]
```diff
@@ -3053,12 +3055,13 @@ export class PokemonSummonData {
...
   public hitCount: integer = 0;
   public endured: boolean = false;
-  public berriesEaten: BerryType[] = [];
+  public berriesEaten: BerryType[] = []; 
 }
 
 export class PokemonBattleSummonData {
   public turnCount: integer = 1;
   public moveHistory: TurnMove[] = [];
+  public abilityTriggeredThisSwitch: boolean = false;
```
Comment (Flashfyre) [inline src/field/pokemon.ts:3064]:
```suggestion
  public abilityTriggered: boolean = false;
```
Being in battle summon data, I believe 'this switch' is implied

File: src/data/ability.ts [hunk 3]
```diff
@@ -882,6 +882,27 @@ export class VariableMovePowerAbAttr extends PreAttackAbAttr {
   }
 }
 
+export class PreAttackChangeType  extends PreAttackAbAttr{
+  private condition: PokemonAttackCondition;
+  constructor (condition:PokemonAttackCondition){
+    super(true);
+    this.condition = condition;
+  }
+  applyPreAttack(pokemon:Pokemon,passive:boolean,defender:Pokemon,move:PokemonMove): boolean{
```
Comment (Flashfyre) [inline src/data/ability.ts:891]:
```suggestion
  private condition: PokemonAttackCondition;
  
  constructor (condition:PokemonAttackCondition) {
    super(true);
    this.condition = condition;
  }
  
  applyPreAttack(pokemon:Pokemon,passive:boolean,defender:Pokemon,move:PokemonMove): boolean{
```

File: src/data/ability.ts [hunk 4]
```diff
@@ -882,6 +882,27 @@ export class VariableMovePowerAbAttr extends PreAttackAbAttr {
...
+        pokemon.summonData.types = [MoveType];
+        pokemon.battleSummonData.abilityTriggeredThisSwitch = true;
+        pokemon.updateInfo();
+        pokemon.scene.queueMessage(getPokemonMessage(pokemon, ` tranformed\ninto the ${Utils.toReadableString(Type[MoveType])} type!`));
+      }
+    }
+    return false;
+    }
+ }
 export class VariableMoveTypeAbAttr extends AbAttr {
```
Comment (Flashfyre) [inline src/data/ability.ts:906]:
```suggestion
  }
}
 
export class VariableMoveTypeAbAttr extends AbAttr {
```

**Review by Flashfyre — CHANGES_REQUESTED** [review]

**Review by takeyourt1me — COMMENTED** [review]
Changed to fit the new suggested name and got rid of that extra line in the ignore. Is there anything else I should do?

File: src/data/ability.ts [hunk 5]
```diff
@@ -3250,7 +3274,7 @@ export function initAbilities() {
     new Ability(Abilities.DAUNTLESS_SHIELD, 8)
       .attr(PostSummonStatChangeAbAttr, BattleStat.DEF, 1, true),
     new Ability(Abilities.LIBERO, 8)
-      .unimplemented(),
+    .attr(PreAttackChangeType,(user,target,move) => move.name !== 'Struggle' && !user.isTerastallized() && user.battleSummonData.abilityTriggered === false),
```
Comment (Flashfyre) [inline src/data/ability.ts:3277]: Checking move name will not work with localization, use move ID

**Review by Flashfyre — CHANGES_REQUESTED** [review]

**Comment by bennybroseph** [comment]
Closing this due to inactivity. Feel free to make a new request if interest is reignited

**Pull request closed without merging**